File: sdk/nodejs/src/resources/hosting.js
"use strict";

const { BaseResource } = require("./base");

class HostingResource extends BaseResource {
  async getReinstall(serviceId) {
    return this._http.get("$/api/service/{id}/reinstall");
  }

  async reinstall(serviceId, data) {
    return this._http.post("$/api/service/{id}/reinstall", data);
  }

  async getStatus(serviceId) {
    return this._http.get("$/api/service/{id}/status");
  }

  async getBandwidthGraphs(serviceId) {
    return this._http.get("$/api/service/{id}/bandwidth-graphs");
  }

  async getBandwidth(serviceId) {
    return this._http.get("$/api/service/{id}/bandwidth");
  }

  async resetPassword(serviceId) {
    return this._http.post("$/api/service/{id}/reset");
  }

}

module.exports = { HostingResource };

File: sdk/java/src/main/java/vn/hitechcloud/sdk/resources/Hosting.java
package vn.hitechcloud.sdk.resources;

import java.util.Map;
import vn.hitechcloud.sdk.HttpClient;

public class Hosting extends BaseResource {

    public Hosting(HttpClient http) {
        super(http);
    }

    /** Get reinstall info */
    public Map<String, Object> reinstall(int id) throws Exception {
        return http.get(String.format("/api/service/%id/reinstall", id));
    }

    /** Reinstall */
    public Map<String, Object> doReinstall(int id, Map<String, Object> data) throws Exception {
        return http.post(String.format("/api/service/%id/reinstall", id), data);
    }

    /** Get status */
    public Map<String, Object> status(int id) throws Exception {
        return http.get(String.format("/api/service/%id/status", id));
    }

    /** Get bandwidth graphs */
    public Map<String, Object> bandwidthGraphs(int id) throws Exception {
        return http.get(String.format("/api/service/%id/bandwidth-graphs", id));
    }

    /** Get bandwidth */
    public Map<String, Object> bandwidth(int id) throws Exception {
        return http.get(String.format("/api/service/%id/bandwidth", id));
    }

    /** Reset password */
    public Map<String, Object> resetPassword(int id, Map<String, Object> data) throws Exception {
        return http.post(String.format("/api/service/%id/reset", id), data);
    }

}
